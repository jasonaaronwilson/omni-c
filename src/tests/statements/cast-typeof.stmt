cast(typeof(value_alist_t**), NULL);

