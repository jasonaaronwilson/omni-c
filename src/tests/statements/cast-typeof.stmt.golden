{
    tag = PARSE_NODE_EXPRESSION_STATEMENT
    expression = {
        tag = PARSE_NODE_OPERATOR
        operator = cast
        left = {
            tag = PARSE_NODE_TYPE
            type_node_kind = TYPE_NODE_KIND_TYPEOF
            user_type = {
                tag = PARSE_NODE_TYPE
                type_node_kind = TYPE_NODE_KIND_POINTER
                type_args = [
                    {
                        tag = PARSE_NODE_TYPE
                        type_node_kind = TYPE_NODE_KIND_POINTER
                        type_args = [
                            {
                                tag = PARSE_NODE_TYPE
                                type_node_kind = TYPE_NODE_KIND_TYPENAME
                                type_name = value_alist_t
                                type_args = [
                                ]
                            }
                        ]
                    }
                ]
            }
            type_args = [
            ]
        }
        right = {
            tag = PARSE_NODE_IDENTIFIER
            token = NULL
        }
    }
}

// C Output
(/*CAST*/(typeof(value_alist_t**)) NULL);

